guess.js

```diff
@@ -0,0 +1,159 @@
+function randomNum(min, max) {
+    return Math.floor(Math.random() * (max - min + 1) + min)
+  }
+
+const readline = require('readline')
+const rli = readline.createInterface(process.stdin, process.stdout)
+
+function capitalize(s){
+    return s.charAt(0).toUpperCase() + s.slice(1).toLowerCase()
+}
+
+function ask(questionText){
+    return new Promise((resolve, reject) => {
+        rli.question(questionText, resolve)
+    })
+}
+
+async function binary_search(min_num, max_num){                                                 // Since I know I'm not looking through an array but a range, I only need to worry about 1 to n
+    let guessNum = Math.floor((min_num + max_num)/2)
+    let response = await ask(`I guess ${guessNum}. Is this number right, too high, or too low? (r/h/l)`)
+    if((response === 'l' || response === 'h') && (max_num - min_num < 2)){
+        console.log("Cheater detected!")
+        process.exit()
+    }
+    switch(response){                                                                       // Unless the input is 'r', I use recursion on this function to loop to the start depending on the response with the new max or min determined by the last max or min
+        case 'r':
+            console.log('Yes! I win!')
+            process.exit()
+        case 'h':
+            console.log(`Hmm... I\'ll remember the number is less than ${guessNum}...`)
+            binary_search(1,guessNum-1)

```

Comment:
Having the low number set to a variable rather then 1 would allow you to adjust the low end and make the computer guesses more efficient